tf2_ros/src/static_transform_broadcaster_node.cpp

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright (c) 2019, Open Source Robotics Foundation, Inc.
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ *     * Redistributions of source code must retain the above copyright
+ *       notice, this list of conditions and the following disclaimer.
+ *     * Redistributions in binary form must reproduce the above copyright
+ *       notice, this list of conditions and the following disclaimer in the
+ *       documentation and/or other materials provided with the distribution.
+ *     * Neither the name of the Willow Garage, Inc. nor the names of its
+ *       contributors may be used to endorse or promote products derived from
+ *       this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include <utility>
+#include <memory>
+#include <stdexcept>
+#include <string>
+#include <random>
+
+#include "tf2_ros/static_transform_broadcaster_node.hpp"
+#include "rclcpp/rclcpp.hpp"
+
+namespace
+{
+constexpr size_t uuid_alpha_count = 8;
+constexpr size_t uuid_numerics_count = 8;
+constexpr size_t uuid_size = uuid_alpha_count + uuid_numerics_count;
+using UUID = std::array<uint8_t, uuid_size>;
+
+std::string get_unique_node_name()
+{
+  std::string node_name{"static_transform_publisher_"};
+  std::random_device rd;
+  std::mt19937 g{rd()};
+  std::uniform_int_distribution<char> alpha{'A', 'Z'};
+  std::uniform_int_distribution<char> numerics{'0', '9'};
+  UUID node_uuid;
+  std::generate(node_uuid.begin(), node_uuid.begin() + uuid_alpha_count, [&alpha, &rd](){ return alpha(rd); });
+  std::generate(node_uuid.begin() + uuid_alpha_count, node_uuid.end(), [&numerics, &rd](){ return numerics(rd); });
+  std::shuffle(node_uuid.begin(), node_uuid.end(), g);  // mix alpha and numerics
+  std::for_each(node_uuid.begin(), node_uuid.end(), [&node_name](auto & byte){ node_name += (byte) ; });
+  return node_name;
```

Comment:
@clalancette I'm not super proud of this, but it works... Do you have any other recommendations?

Reply:
@allenh1 It's OK, but I feel like it is overly complicated and not performant enough (the latter doesn't matter *too* much here, but still).  I played around with this a bit and came up with: https://gist.github.com/clalancette/fedda4179d8956a13bb4e782f8c09dbc ; what do you think about that?

Reply:
>  I feel like it is overly complicated and not performant enough

Yep. 100% agreed.

> what do you think about that?

I like your implementation better. Swapped it in.